File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from datetime import datetime

class MyNode(Node):
    def __init__(self) -> None:
        super().__init__("py_test")
        self.__counter = 0
        self.get_logger().info("Hello ROS2")
        time = datetime.now().strftime("%H:%M:%S")
        self.get_logger().info(f"Then time is {time}")
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self) -> None:
        self.__counter += 1
        self.get_logger().info(f"Hello {self.__counter}")

def main(args = None):
    rclpy.init(args = args)
    node = Node("py_test")
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

using namespace rclcpp;

class MyNode : public Node
{
public:
    MyNode() : Node("cpp_test"), _counter(0)
    {
        RCLCPP_INFO(this->get_logger(), "Hello Cpp Node");

        _timer = this->create_wall_timer(std::chrono::seconds(1),
                                         std::bind(&MyNode::timerCallback, this));
    }

protected:
    // NOP

private:
    void timerCallback()
    {
        _counter++;
        RCLCPP_INFO(this->get_logger(), "Hello %d", _counter);
    }

    TimerBase::SharedPtr _timer;
    int _counter;
};

int main(int argc, char **argv)
{
    init(argc, argv);   // Init Ros2
    auto node = std::make_shared<MyNode>();  // Create a node
    RCLCPP_INFO(node->get_logger(), "Hello Cpp Node"); // Made the node print
    spin(node);                                        // call back on the node
    shutdown();                                        // shutdown comm
    return 0;
}